lib/Optimize.cpp

```diff
@@ -875,6 +874,48 @@ static void ReplaceMemWriteOp(const Program &program, llvm::Module &module,
   RemoveFunction(func);
 }
 
+// Lower an anvill type function into an `inttoptr` instructionz
+static void ReplaceTypeOp(const Program &program, llvm::Module &module,
+                          const std::string &name, llvm::Type *val_type) {
+  auto func = module.getFunction(name);
+  if (!func) {
+    LOG(ERROR) << "Function " << name << " not found";
+    return;
+  }
+  CHECK(func->isDeclaration())
+      << "Cannot lower already implemented memory intrinsic " << name;
+  auto callers = remill::CallersOf(func);
+  for (auto call_inst : callers) {
+    auto arg_val = call_inst->getArgOperand(0);
+    llvm::IRBuilder<> irb(call_inst);
+
+    //Assuming that the addr value is supposed to be 0, and that arg_val is a subsitute for addr.
+    llvm::Value *ptr = GetPointer(program, module, irb, arg_val, val_type, 0);
+
+    //The ptr value should be the return type of the function, which is the binary ninja type.
+    //Replace the call with uses of this pointer value
+    call_inst->replaceAllUsesWith(ptr);
+  }
+  //Clean up
+  for (auto call_inst : callers) {
+    call_inst->eraseFromParent();
+  }
+  RemoveFunction(func);
+}
+
+static void LowerTypeOps(const Program &program, llvm::Module &mod) {
+  std::vector<llvm::Function *> funcs;
+  for (auto &func : mod) {
+    funcs.push_back(&func);
+  }
+  for (auto &func : funcs) {
+    if (func->hasName() && func->getName().startswith("__anvill_type")) {
+      ReplaceTypeOp(program, mod, std::string(func->getName()),
```

Comment:
Here you have an `llvm::Function *`, but then ask for the name, pass the name in, then in the `ReplaceTypeOp` function by looking up the function by name. Why not just pass in the `llvm::Function *`? Then the error checking on whether or not the function can be found can be removed, and you can move the return type stuff into the above function.